unix/vncpasswd/vncpasswd.cxx

```diff
@@ -165,24 +165,40 @@ int main(int argc, char** argv)
     FILE* fp = fopen(fname,"w");
     if (!fp) {
       fprintf(stderr,"Couldn't open %s for writing\n",fname);
+      delete [] fname;
+      delete obfuscated;
+      if (obfuscatedReadOnly)
+        delete obfuscatedReadOnly;
       exit(1);
     }
     chmod(fname, S_IRUSR|S_IWUSR);
 
     if (fwrite(obfuscated->buf, obfuscated->length, 1, fp) != 1) {
       fprintf(stderr,"Writing to %s failed\n",fname);
+      delete [] fname;
+      delete obfuscated;
+      if (obfuscatedReadOnly)
+        delete obfuscatedReadOnly;
       exit(1);
     }
 
     if (obfuscatedReadOnly) {
       if (fwrite(obfuscatedReadOnly->buf, obfuscatedReadOnly->length, 1, fp) != 1) {
         fprintf(stderr,"Writing to %s failed\n",fname);
+        delete [] fname;
+        delete obfuscated;
```

Comment:
Delete these as soon as they are no longer need to avoid having to repeat this constantly.

Reply:
fname is used on the debug output, cannot be deleted earlier.